ILSpy/Commands/ExtractPackageEntryContextMenuEntry.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) 2021 Siegfried Pammer
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this
+// software and associated documentation files (the "Software"), to deal in the Software
+// without restriction, including without limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons
+// to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+// PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+// FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using ICSharpCode.Decompiler;
+using ICSharpCode.ILSpy.Properties;
+using ICSharpCode.ILSpy.TextView;
+using ICSharpCode.ILSpy.TreeNodes;
+
+using Microsoft.Win32;
+
+namespace ICSharpCode.ILSpy
+{
+	[ExportContextMenuEntry(Header = nameof(Resources.ExtractPackageEntry), Category = nameof(Resources.Save), Icon = "Images/Save")]
+	sealed class ExtractPackageEntryContextMenuEntry : IContextMenuEntry
+	{
+		public void Execute(TextViewContext context)
+		{
+			// Get all assemblies in the selection that are stored inside a package.
+			var selectedNodes = context.SelectedTreeNodes.OfType<AssemblyTreeNode>()
+				.Where(asm => asm.PackageEntry != null).ToArray();
+			// Get root assembly to infer the initial directory for the save dialog.
+			var bundleNode = selectedNodes.FirstOrDefault()?.Ancestors().OfType<AssemblyTreeNode>()
+				.FirstOrDefault(asm => asm.PackageEntry == null);
+			if (bundleNode == null)
+				return;
+			var assembly = selectedNodes[0].PackageEntry;
```

Comment:
`IsVisible` only checks that at least one bundle member is selected; there's no guarantee that `selectedNodes[0]` is one.